core/services/functions/external_adapter_client.go

```diff
@@ -0,0 +1,94 @@
+package functions
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+
+	"github.com/pkg/errors"
+
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+type ExternalAdapterInterface struct {
+	AdapterURL url.URL
+}
+
+type Payload struct {
+	RequestId            string `json:"requestId"`
+	JobName              string `json:"jobName"`
+	EncryptedSecretsUrls string `json:"encryptedSecretsUrls"`
+}
+
+type Response struct {
+	Result     string       `json:"result"`
+	Data       ResponseData `json:"data"`
+	StatusCode int          `json:"statusCode"`
+}
+
+type ResponseData struct {
+	Result      string `json:"result"`
+	Error       string `json:"error"`
+	ErrorString string `json:"errorString"`
+}
+
+func (ea ExternalAdapterInterface) FetchEncryptedSecrets(ctx context.Context, encryptedSecretsUrls []byte, requestId string, jobName string) (encryptedSecrets, userError []byte, err error) {
```

Comment:
It looks like the only thing specific to secrets is the Payload struct. Can we convert most of this method into a generic request method and only have a small specialized one for secrets on top of it?
We will likely reuse the generic one for the other bridge soon.